malariagen_data/ag3.py

```diff
@@ -2489,6 +2490,92 @@ def aa_allele_frequencies(
 
         return df_aaf
 
+    def plot_frequencies_heatmap(
+        self, df, index, index_name, max_len=100, width=500, colorbar=True
+    ):
+
+        """Generate a Plotly.express frequency heatmap using pandas DataFrame output from ag3.snp_allele_frequencies()
+        or ag3.gene_cnv_frequencies() that has been user-filtered to just rows of interest (fewer rows than max_len).
+
+        Parameters
+        ----------
+        df : pandas DataFrame
+           Filtered output from ag3.snp_allele_frequencies() or ag3.gene_cnv_frequencies.
+        index : list of str
+            A list of one or more column headers present in the input dataframe, these become
+            the heatmap y-axis row labels. When taken together, these columns must produce a unique index.
+        index_name : str
+            This is the y-axis label that will be displayed on the heatmap
+        max_len : int, optional
+            Displaying large styled dataframes may cause ipython notebooks to crash.
+        width : int
```

Comment:
It would be good to make this the width per column, so the size of the dataframe scales naturally if there are more or less columns. So to get the actual width you pass to plotly, multiple the `width` parameter passed by the user by the number of frequency columns.

Reply:
If I understand correctly, the colorbar/row labels/_etc._ are included in "width", so it isn't just a multiple of columns that we want, plus colorbar and row labels are moveable feasts size-wise. I guess the work around is default to no width, but use it if given?